aws/resource_aws_instance.go

```diff
@@ -119,6 +119,14 @@ func resourceAwsInstance() *schema.Resource {
 				Optional:      true,
 				ForceNew:      true,
 				ConflictsWith: []string{"user_data_base64"},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// Sometimes the EC2 API responds with the equivalent, empty SHA1 sum
+					// echo -n "" | shasum
+					if old == "da39a3ee5e6b4b0d3255bfef95601890afd80709" && new == "" {
```

Comment:
@bflad  can we also do the check on the other direction?

If my plan used to omit the `user_data` field but now has to include it with values that are sometimes empty, I would like the diff to be suppressed, too.

```
if (old == "da39a3ee5e6b4b0d3255bfef95601890afd80709" && new == "") ||
   (old == "" && new == "da39a3ee5e6b4b0d3255bfef95601890afd80709") {
```

Reply:
could not wait and created #5467